version: "3.8"
services:
  query:
    build:
      context: .
      dockerfile: "Dockerfile_Consultas"
    container_name: "reportes"
    depends_on:
      - app
    volumes:
      - .:/Users/Agustin/Documents/Cloud
    command: ["python3", "consultas.py"]
  app:
    build:
      context: .
      dockerfile: "Dockerfile"
    container_name: "creodb"
    depends_on:
      db:
        condition: service_healthy 
  db:
    image: postgres
    volumes:
      - ./tabla.sh:/docker-entrypoint-initdb.d/tabla.sh            
    container_name: "db1"
    restart: unless-stopped
    ports:
      - "5432:5432"
    expose:
      - "5432"
    environment:
      POSTGRES_PASSWORD: postgres
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      timeout: 20s
      retries: 10
